packages/EasyErrorHandler/src/Bridge/Bugsnag/BugsnagReporter.php

```diff
@@ -52,12 +51,20 @@ public function __construct(
     public function report(Throwable $throwable)
     {
         $exceptionClass = \get_class($throwable);
-        foreach ($this->ignoreExceptions as $ignoreClass) {
-            if (\is_a($exceptionClass, $ignoreClass, true)) {
-                return;
+        if (\is_array($this->ignoreExceptions) === true) {
+            foreach ($this->ignoreExceptions as $ignoreClass) {
+                if (\is_a($exceptionClass, $ignoreClass, true)) {
+                    return;
+                }
             }
         }
 
+        if ($this->ignoreExceptionsResolver !== null
+            && \is_a($this->ignoreExceptionsResolver, BugsnagIgnoreExceptionsResolverInterface::class, true) === true
+            && $this->ignoreExceptionsResolver::shouldIgnore($throwable) === true) {
```

Comment:
I like the idea of delegating the decision to an object as it will allow applications to extend this logic to fit their needs. However, I don't like the current implementation, defining a config pointing to a class with a static method seems complicated and not really straight forward, and on the top of that it restricts the possibilities of the applications how to implement their logic.

It would be much better to use the "service approach" for this:

- create a dedicated service to decide if an exception should be ignored in bugsnag
- inject this service in this BugsnagReporter
- create a default implementation for it (could return false by default)

It will simplify the logic in the reporter as it will always expect to have an instance of this service to make the decision.
It will simplify the process for applications to extend/replace the default logic by decorating/replacing the service, AND will allow them to use all the dependency injection features to implement their own logic!

- no need for new config and param
- simpler logic in bugsnag reporter
- unlimited possibilites for bugsnag ignore logic

Reply:
Refactored a little. Please check it